### PR TITLE
test: init: convert to dir helpers

### PR DESCRIPTION
**Pull request by efiop**
Related to #2896 
* [x] ❗ Have you followed the guidelines in the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) list?

* [x] 📖 Check this box if this PR **does not** require [documentation](https://dvc.org/doc) updates, or if it does **and** you have created a separate PR in [dvc.org](https://github.com/iterative/dvc.org) with such updates (or at least opened an issue about it in that repo). Please link below to your PR (or issue) in the [dvc.org](https://github.com/iterative/dvc.org) repo.

* [x] ❌ Have you checked DeepSource, CodeClimate, and other sanity checks below? We consider their findings recommendatory and don't expect everything to be addressed. Please review them carefully and fix those that actually improve code or fix bugs.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: tests/func/test_init.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def test_cli(self):
         self._test_init()
 
 
-def test_init_quiet_should_not_display_welcome_screen(git, caplog):
+def test_init_quiet_should_not_display_welcome_screen(scm, caplog):
```
Comment (jorgeorpinel) [inline tests/func/test_init.py]: This remind me of iterative/dvc.org/issues/540 though.

Comment (efiop) [reply tests/func/test_init.py]: @jorgeorpinel Yeah, me too. Maybe we will rename it to git after the migration is done.

Comment (Suor) [reply tests/func/test_init.py]: We probably won't. People will assume `git` being some gitpython instance.

Comment (Suor) [inline tests/func/test_init.py]: Let's use `tmp_dir` fixture explicitly.

Comment (efiop) [reply tests/func/test_init.py]: @Suor Why though?

Comment (efiop) [reply tests/func/test_init.py]: scm creates a temp dir anyways, I see no good reason to put `tmp_dir` here. Esp since it is so obvious.

Comment (Suor) [reply tests/func/test_init.py]: 1. We do it everywhere else.
2. It wasn't obvious for me, I needed to check the code to be sure it won't just `git init` in curdir.

Comment (efiop) [inline tests/func/test_init.py]:
```suggestion
def test_init_quiet_should_not_display_welcome_screen(tmp_dir, scm, caplog):
```

Comment (efiop) [reply tests/func/test_init.py]: Added.

**Pull request merged**